src/core/reducers/authentication.js

```diff
@@ -1,9 +1,15 @@
+import cookie from 'react-cookie';
+import config from 'config';
+
 export default function authentication(state = {}, action) {
   const { payload, type } = action;
   if (type === 'SET_JWT') {
-    return { token: payload.token };
+    return { ...state, token: payload.token };
   } else if (type === 'SET_CURRENT_USER') {
     return { ...state, username: payload.username };
+  } else if (type === 'LOG_OUT_USER') {
+    cookie.remove(config.get('cookieName'));
```

Comment:
We're currently taking the cookie from AMO and setting it on the admin domain through JS. I think we could potentially set `httpOnly` from the client.

We could set it as `httpOnly` and then add an endpoint to addons-frontend to remove it on DELETE. Since we are including the JWT in the page that might not help much. It would probably be better to leave `httpOnly` off and read from the cookie and remove it from the page.

This is tricky because we don't want the JWT to be read on an XSS but we want to be able to read it ourselved so I don't think there's much we can do. Reading the cookie might be the best since then it isn't included in the source if someone saves the HTML for some reason.